### PR TITLE
xc7: skip prohibited sites during physical tiles import

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

This PR adds the possibility to skip the physical tile import in the DB whenever the tiles contain prohibited sites.

This requires first https://github.com/SymbiFlow/prjxray/pull/1278

### CODE REVIEW

**Comment by litghost** [comment]
prjxray change has been merged, add submodule update?

File: utils/lib/connection_database.sql [hunk 1]
```diff
@@ -81,6 +81,7 @@ CREATE TABLE phy_tile(
   grid_x INT,
   grid_y INT,
   clock_region_pkey INT,
+  prohibited INT,
```
Comment (litghost) [inline utils/lib/connection_database.sql]: Sites are prohibited, not tiles.  This is the wrong place for the flag.  Also this should be a BOOLEAN, not an INT.

Comment (acomodi) [reply utils/lib/connection_database.sql]: The only reason to set the prohibited to Tiles is that it was more straightforward to skip the tile itself during arch_import, but I guess it is more consistent as you suggested. Will fix that and bump prjxray

File: xc/common/utils/prjxray_arch_import.py [hunk 2]
```diff
@@ -480,6 +480,18 @@ def get_tiles(
...
+        if phy_tile_pkey is not None:
+            c2.execute(
+                "SELECT prohibited FROM site_instance WHERE phy_tile_pkey = ?",
+                (phy_tile_pkey, )
+            )
+
+            prohibited_sites = c2.fetchall()
+
+            # Skip generation of prohibited tiles
+            if True in prohibited_sites:
```
Comment (litghost) [inline xc/common/utils/prjxray_arch_import.py]: This won't work as expected.  fetchall returns [(True), (True)] not [True, True].

Comment (litghost) [reply xc/common/utils/prjxray_arch_import.py]: What I've done is unpack the tuple at fetchall:
```
any_prohibited_sites = any(prohibited for (prohibited,) in c2.fetchall())
```

Comment (acomodi) [reply xc/common/utils/prjxray_arch_import.py]: Yep, much cleaner, thanks for the suggestion.

File: utils/lib/connection_database.sql (hunk 1 above)
Comment (acomodi) [reply utils/lib/connection_database.sql]: Fixed

File: xc/common/utils/prjxray_arch_import.py (hunk 2 above)
Comment (acomodi) [reply xc/common/utils/prjxray_arch_import.py]: Fixed

File: xc/common/utils/prjxray_form_channels.py [hunk 3]
```diff
@@ -578,6 +582,15 @@ def import_phy_grid(db, grid, conn, get_switch, get_switch_timing):
                 )
             )
 
+            site_instance_pkey = write_cur.lastrowid
+
+            cur.execute(
```
Comment (litghost) [inline xc/common/utils/prjxray_form_channels.py]: This should be "write_cur" not "cur", and then line 490 is not required.

Comment (acomodi) [reply xc/common/utils/prjxray_form_channels.py]: Sure, forgot about that. Done

**Comment by acomodi** [comment]
@litghost Seems that thye Doc and Testing Arch CIs are still running, but by opening them they show up to be complete (at least to me). Would be ok to merge?

**Pull request merged**